spec/system/consumer/shopping/products_spec.rb

```diff
@@ -47,16 +47,19 @@
         add_variant_to_order_cycle(exchange1, variant)
       end
 
-      it "shows HTML product description" do
-        product.description = '<p><b>Formatted</b> product description.</p> Link to an '\
+      it "shows HTML product description and truncates it correctly" do
+        pending "#10685"
+        product.description = '<p><b>Formatted</b> product description: Lorem ipsum dolor sit amet,
+                              consectetur adipiscing elit. Morbi venenatis metus diam,
+                               eget scelerisque nibh auctor non. </p> Link to an '\
                               '<a href="http://google.fr" target="_blank">external site</a>'\
                               '<img src="https://www.openfoodnetwork.org/wp-content/uploads/2019/'\
                               '05/logo-ofn-global-web@2x.png" alt="open food network logo" />'
         product.save!
 
         visit shop_path
         expect(page).to have_content product.name
-
+        expect(page).to have_content "..." # it truncates a long produt description
```

Comment:
I'd add a `withing`, something like (not tested):
```ruby
within "product#product-#{product.id}"
  expect("product-description").to have_content "..."
end
```
to be more specific. 


What do you think?

Reply:
Fully agree - looking for three loose dots on the whole page is probably not a very good idea :+1: 

Maybe even adding it within the helper `expect_product_description_html_to_be_displayed` ?
Proposal coming up.